cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/SideEffects.qll

```diff
@@ -111,6 +111,32 @@ private predicate hasDefaultSideEffect(Call call, ParameterIndex i, boolean buff
   )
 }
 
+/**
+ * Gets the `SideEffectFunction` called by the specified expression.class, if known.
+ *
+ * Thie will return a result only for `Call`, in which case it returns the target of the call, or
+ * for `NewExpr` and `NewArrayExpr`, in which case it returns the allocator function called by the
+ * expression.
+ */
+private SideEffectFunction getCallOrAllocationSideEffectFunction(Expr expr) {
+  result = expr.(Call).getTarget()
+  or
+  result = expr.(NewOrNewArrayExpr).getAllocator()
+}
+
+/**
+ * Returns the side effect opcode, if any, that represents any side effects not specifically modeled
+ * by an argument side effect.
+ */
+Opcode getCallSideEffectOpcode(Expr expr) {
```

Comment:
This predicate doesn't bind `expr` at all, does it (i.e., because of the negation)?

Reply:
I modified that predicate in a subsequent commit. Take a look at the changes from all commits for `SideEffects.qll` and see what you think.

Reply:
Oh! Indeed, that looks like the proper fix.